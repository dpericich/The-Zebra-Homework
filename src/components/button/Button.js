import React from "react";
// SVG files for the different logos
import Bolt from "../../assets/IconBoltWhite.png";
import Arrow from "../../assets/IconArrowDown.png";

const Button = ({ text, type, link }) => {
  // Determine which icon to load based on the type passed in
  let logo;
  if (Number(type) === 0 || Number(type) === 1) {
    logo = Arrow;
  } else if (Number(type) === 2) {
    logo = null;
  } else if (Number(type) === 3) {
    logo = Bolt;
    text = "Buy Now";
  }

  let color;
  if (type === 0 || type === 1) {
    color = "blue";
  } else if (type === 3) {
    color = "orange";
  } else {
    color = "";
  }

  if (type !== 2) {
    return (
      <div className={`button button__${color}`}>
        <a href={link} className="button__link">
          <img src={logo} alt="quote button" className="button__logo" />
          <div className="button__text">{text || null}</div>
        </a>
      </div>
    );
  } else {
    return null;
  }
};

export default Button;
