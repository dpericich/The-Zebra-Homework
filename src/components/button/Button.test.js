import React, { Component } from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Button from "./Button";
import { getByText } from "@testing-library/react";

describe("<Button />", () => {
  test("it renders without issue", () => {
    const component = shallow(<Button />);
    expect(toJSON(component)).toMatchSnapshot();
  });
  test("it renders correctly with type of 1", () => {
    const type = 1;
    const component = shallow(<Button type={type} />);
    expect(toJSON(component)).toMatchSnapshot();
  });
  test("it renders correctly with type of 2", () => {
    const type = 2;
    const component = shallow(<Button type={type} />);
    expect(toJSON(component)).toMatchSnapshot();
  });
  test("it renders correctly with type 3", () => {
    const type = 3;
    const component = shallow(<Button type={type} />);
    expect(toJSON(component)).toMatchSnapshot();
  });
});
