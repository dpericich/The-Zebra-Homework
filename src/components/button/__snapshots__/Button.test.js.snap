// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it renders correctly with type 3 1`] = `
<div
  className="button button__orange"
>
  <a
    className="button__link"
  >
    <img
      alt="quote button"
      className="button__logo"
      src="IconBoltWhite.png"
    />
    <div
      className="button__text"
    >
      Buy Now
    </div>
  </a>
</div>
`;

exports[`<Button /> it renders correctly with type of 1 1`] = `
<div
  className="button button__blue"
>
  <a
    className="button__link"
  >
    <img
      alt="quote button"
      className="button__logo"
      src="IconArrowDown.png"
    />
    <div
      className="button__text"
    />
  </a>
</div>
`;

exports[`<Button /> it renders correctly with type of 2 1`] = `""`;

exports[`<Button /> it renders without issue 1`] = `
<div
  className="button button__"
>
  <a
    className="button__link"
  >
    <img
      alt="quote button"
      className="button__logo"
    />
    <div
      className="button__text"
    />
  </a>
</div>
`;
